attest/verifier/src/status.rs

```diff
@@ -34,7 +34,7 @@ fn check_ids(quote_status: &IasQuoteResult, config_ids: &[String], sw_ids: &[Str
         }
         Err(IasQuoteError::ConfigurationAndSwHardeningNeeded { advisory_ids, .. }) => advisory_ids
             .iter()
-            .all(|id| config_ids.contains(id) || sw_ids.contains(id)),
+            .all(|id| config_ids.contains(id) && sw_ids.contains(id)),
```

Comment:
This is an important change: due to the way Intel is handling 00615, the original logic here is incorrect.

Previously, if you got `CONFIGURATION_AND_SW_HARDENING_NEEDED`, having the relevant advisory in either config_ids or sw_ids would be sufficient to pass.

With INTEL-SA-00615, IAS will return a `CASH_N` for 00615 if hyperthreading is enabled, otherwise it will return just `SW_HARDENING_NEEDED`. We want to allow `SH_N` + 00615, but not `CASH_N` + 00615, so we make our `CASH_N` handler check require the advisory in both lists.

This makes some other cases (which we don't currently use) awkward, which is why it wasn't done that way in the beginning.